sycl/test/matrix/matrix-tf32-test.cpp

```diff
@@ -0,0 +1,165 @@
+               sg);
+           joint_matrix<float, TM, TN> sub_c(sg);
+           joint_matrix_load(sg, sub_c,
+                             accC.get_pointer() + (sg_startx * TM) * N +
+                                 sg_starty / SG_SZ * TN,
+                             N, matrix_layout::row_major);
+           for (int k = 0; k < K; k += TK) {
+             joint_matrix_load(sg, sub_a,
+                               accA.get_pointer() + (sg_startx * TM) * K + k, K,
+                               matrix_layout::row_major);
+             // Assume we alreay in vnni format.
+             joint_matrix_load(sg, sub_b,
+                               accB.get_pointer() + (k) * (N) +
+                                   sg_starty / SG_SZ * TN,
+                               N, matrix_layout::packed_b);
+             sub_c = joint_matrix_mad(sg, sub_a, sub_b, sub_c);
+           }
+           auto wi_slice_a = sub_a.get_wi_data();
+           for (int i = 0; i < wi_slice_a.length(); i++) {
+             float elem = wi_slice_a[i];
+             // TODO: OP= is buggy: __spirv_VectorInsertDynamic should take
+             // storage element type as argument
+             // wi_slice_a[i] *= 2;
+           }
+           joint_matrix_store(sg, sub_c,
+                              accC.get_pointer() + (sg_startx * TM) * N +
+                                  sg_starty / SG_SZ * TN,
+                              N, matrix_layout::row_major);
+         }); // parallel for
+   }).wait();
+}
+
+static constexpr size_t MATRIX_M = TM * 2;
+static constexpr size_t MATRIX_N = TN * 2;
+static constexpr size_t MATRIX_K = TK * 2;
+precision::tf32 A[MATRIX_M][MATRIX_K];
+precision::tf32 B[MATRIX_K][MATRIX_N];
+float C[MATRIX_M][MATRIX_N];
+float D[MATRIX_M][MATRIX_N];
+
+// this is a hack and should be replaced with a spirv function
+precision::tf32 make_tf32(float x) {
+  uint32_t y = reinterpret_cast<uint32_t const &>(x);
+  y += 0x1000u;
+  precision::tf32 res = reinterpret_cast<precision::tf32 const &>(y);
+  return res;
+}
+
+void matrix_multiply_ref(float *A_mem, float *B_mem, float *C_mem, int M, int N,
+                         int K) {
+  for (int m = 0; m < M; m++)
+    for (int n = 0; n < N; n++) {
+      for (int k = 0; k < K; k++) {
+        float va = *(float *)(A_mem + m * K + k);
+        float vb = *(float *)(B_mem + k * N + n);
+        float acc = *((float *)(C_mem + m * N + n));
+        *((float *)(C_mem + m * N + n)) = va * vb;
+      }
+    }
+}
+
+int main() {
+  for (int i = 0; i < MATRIX_M; i++) {
+    for (int j = 0; j < MATRIX_K; j++) {
+      A[i][j] = make_tf32(1.0f * (i + j));
+    }
+  }
+  for (int i = 0; i < MATRIX_K / 2; i++) {
+    for (int j = 0; j < MATRIX_N * 2; j++) {
+      B[i][j] = make_tf32(2.0f * i + 3.0f * j);
+    }
+  }
+  for (int i = 0; i < MATRIX_M; i++) {
+    for (int j = 0; j < MATRIX_N; j++) {
+      C[i][j] = 1.0;
+      D[i][j] = 1.0;
+    }
+  }
+
+  big_matrix<float, MATRIX_M, MATRIX_N> MC((float *)&C);
```

Comment:
```suggestion
  big_matrix<float, MATRIX_M, MATRIX_N> MC { C };
```
and add the right constructor in `big_matrix`